openfisca_core/simulations/simulation_builder.py

```diff
@@ -84,11 +119,17 @@ def build_from_entities(self, tax_benefit_system, input_dict):
             )
 
         helpers.check_type(input_dict, dict, ["error"])
-        axes = input_dict.pop("axes", None)
+
+        axes: list[list[AxisParams]] | None = None
+
+        if is_axes_spec(input_dict):
+            axes = input_dict.pop("axes")
+
+        params: NoAxParams = typing.cast(NoAxParams, input_dict)
```

Comment:
This should be named NoAxisParams or NoAxesParams. An Ax is a weapon ;)

I feel a bit queasy about "No" as a suffix. For one thing we have things like NoSQL, NoEstimates and look how well they've turned out ;) Or another thing that comes to mind is how many things are called "wireless" that are not primarily characterized by the absence of wires.

Perhaps something like UnitaryParams would be a better focus on what the type *is* rather than on what it *isn't*.

Reply:
Yeah that is chapter 1 of Uncle Bob's clean code : beware of name encodings. I'll change it to something more semantic.